### PR TITLE
Fix various bugs with the new teambuilder

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

File: src/battle-dex-search.ts [hunk 1]
```diff
@@ -542,8 +542,7 @@ abstract class BattleTypedSearch<T extends SearchType> {
 	 */
 	set: PokemonSet | null = null;
 
-	protected isDoubles = false;
-	protected isLetsGo = false;
+	protected mod: 'natdex' | 'letsgo' | 'doubles' | null = null;
```
Comment (Zarel) [inline src/battle-dex-search.ts:545]: I don't like `mod` as a name for this because it already means something else.

Comment (Zarel) [reply src/battle-dex-search.ts:545]: `formatType`?

File: src/battle-dex-search.ts [hunk 2]
```diff
@@ -520,7 +520,7 @@ class DexSearch {
 abstract class BattleTypedSearch<T extends SearchType> {
 	searchType: T;
 	/**
-	 * Dex for the mod/generation to search.
+	 * Dex for the formatType/generation to search.
```
Comment (Zarel) [inline src/battle-dex-search.ts:523]:
```suggestion
	 * Dex for the mod/generation to search.
```

This one actually means "mod", which is why I didn't want you to do the same name as that. :p

Comment (KrisXV) [reply src/battle-dex-search.ts]: not sure how this still got changed.... i very intentionally skipped this

**Pull request merged**